tests/test_qq_file_fallback.py

```
@@ -0,0 +1,406 @@
+    def test_apk_extension_in_name(self):
+        m = load_fallback_module()
+        assert m._is_apk_component(_file_component(m, name="app.apk")) is True
+
+    def test_xapk_apkm_apks_extensions(self):
+        m = load_fallback_module()
+        for ext in (".xapk", ".apkm", ".apks"):
+            assert m._is_apk_component(_file_component(m, name=f"app{ext}")) is True
+
+    def test_apk_in_source_path_only(self):
+        m = load_fallback_module()
+        comp = _file_component(m, name="renamed.txt", source_path="/tmp/real.apk")
```

Ruff (0.15.20)

[error] Probable insecure usage of temporary file or directory: "/tmp/real.apk"

(S108)